android/greengrass-nucleus-android/src/main/java/com/aws/greengrass/android/provision/BaseProvisionManager.java

```diff
@@ -0,0 +1,452 @@
+                    .resolve(version)
+                    .toString()).getCanonicalFile();
+        } catch (IOException e) {
+            logger.atError().setCause(e).log("Couldn't create a new file.");
+        }
+        String name = DeviceConfiguration.DEFAULT_NUCLEUS_COMPONENT_NAME.toLowerCase();
+
+
+        if (rootUnpackDir != null) {
+            return findDir(rootUnpackDir, name);
+        }
+        return null;
+    }
+
+    /**
+     * Prepare asset files.
+     *
+     * @param context context.
+     */
+    public void prepareAssetFiles(Context context) {
+        Path distro = WorkspaceManager.getCurrentDistroPath();
+        Path distroLink = distro.resolve(CURRENT_DISTRO_LINK);
+
+        if (isFolderEmpty(distroLink)) {
+            copyAssetFiles(context, Paths.get(NUCLEUS_FOLDER));
+            String version = getNucleusVersionFromAssets(context, Paths.get(NUCLEUS_FOLDER));
+
+            try {
+                Utils.createPaths(distro);
+                Files.deleteIfExists(distroLink);
+            } catch (IOException e) {
+                logger.atError().setCause(e).log("Couldn't create folders.");
+            }
+
+            try {
+                Files.createSymbolicLink(distroLink, getCurrentUnpackDir(version));
+            } catch (IOException e) {
+                logger.atError().setCause(e).log("Unable to create symbolic link.");
+            }
+        }
+    }
+
+    /**
+     * Get Nucleus version in Assets.
+     *
+     * @param context context.
+     * @param path root assets path.
+     *
+     * @return Nucleus version.
+     */
+    private String getNucleusVersionFromAssets(@NonNull Context context, Path path) {
+        try {
+            String [] list = context.getAssets().list(path.toString());
+            // Get a folder name - is the Nucleus version
+            for (String file : list) {
+                if (file != null) {
+                    return file;
+                }
+            }
+        } catch (IOException e) {
+            return null;
+        }
+
+        return null;
+    }
+
+    /**
+     * Copy asset files to the ROOT_INIT_FOLDER folder.
+     *
+     * @param context context.
+     * @param path root assets path.
+     *
+     * @return result of copy.
+     */
+    private boolean copyAssetFiles(@NonNull Context context, Path path) {
+        try {
+            String [] list = context.getAssets().list(path.toString());
+            // This is a folder
+            for (String file : list) {
+                if (!copyAssetFiles(context, path.resolve(file))) {
+                    return false;
+                } else {
+                    try {
+                        Path targetPath = WorkspaceManager.getUnarchivedPath().resolve(path);
+                        Utils.createPaths(targetPath);
+
+                        File targetFile = new File(targetPath.resolve(file).toString());
+                        targetFile.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)